testplan/common/utils/match.py

```diff
@@ -296,22 +536,49 @@ def get_between(self, mark1=None, mark2=None):
         :return: The content between mark1 and mark2.
         :rtype: ``str``
         """
-        if mark1 is not None and mark2 is not None:
-            if self.marks[mark1] >= self.marks[mark2]:
-                raise ValueError(
-                    'Mark "{}" must be present before mark "{}"'.format(
-                        mark1, mark2
+        try:
+            log_handle = None
+
+            if mark1 is not None and mark2 is not None:
+                if self.strategy.validate_marks(
+                    mark1=self.marks[mark1], mark2=self.marks[mark2]
+                ):
+                    raise ValueError(
+                        f'Mark "{mark2}" must be present before mark "{mark1}"'
                     )
-                )
 
-        with open(self.log_path, "r") as log:
-            start_pos = self.marks[mark1] if mark1 is not None else 0
-            end_pos = self.marks[mark2] if mark2 is not None else None
-            log.seek(start_pos)
-            if not end_pos:
-                return log.read()
             lines_between = []
-            while end_pos > log.tell():
-                line = log.readline()
-                lines_between.append(line)
+            start_position = (
+                self.marks[mark1]
+                if mark1 is not None
+                else self.strategy.get_start_of_file_position()
+            )
+
+            end_position = (
+                self.marks[mark2]
+                if mark2 is not None
+                else self.strategy.get_end_of_file_position()
+            )
+
+            log_handle = self.strategy.open_logfile_in_position(
+                position=start_position, mode="r"
+            )
+
+            while self.strategy.end_position_reached(
+                end_position=end_position, file_handle=log_handle
+            ):
+                line = log_handle.readline()
+
+                if line:
+                    lines_between.append(line)
+                else:
+                    log_handle = self.strategy.get_next_file(
```

Comment:
old log handle need to be closed

Reply:
For the NoLogRotationStrategy I will be returning the same handle so I am not closing it there. But in the other strategy types that require rotation, the old file handle is closed in the **get_next_file** function before returning the new log handle.